core/src/main/java/io/methvin/watcher/DirectoryWatcher.java

```diff
@@ -224,15 +242,28 @@ public DirectoryWatcher(
    * is closed.
    */
   public void watch() {
+    final AtomicInteger eventCount = new AtomicInteger(0);
```

Comment:
Why is this atomic? the main watch() loop  is single threaded, no other thread reads this value. It's just a value that is passed to the event handler.

Reply:
yeap, agree